[Mail] add service mailhog to docker-compose

Moemen-Gaballah opened the pull request:
| Q             | A                                                                  |
|---------------|--------------------------------------------------------------------|
| Branch?       | 1.x <!-- see below -->                                             |
| Bug fix?      | no                                                             |
| New feature?  | no <!-- please update /CHANGELOG.md files -->                  |
| Deprecations? | no <!-- please update UPGRADE-*.md and /CHANGELOG.md files --> |
| Tickets       | Fix #289  <!-- prefix each issue number with "Fix #", -->           |
| License       | MIT                                                                |

imdhemy reviewed (changes requested):
This container is supposed to be used only for development.

`docker-compose.yml` file is used for production.
`docker-compose.override.yml` file is used for development.

Moemen-Gaballah commented:
@imdhemy 


> This container is supposed to be used only for development.
> 
> `docker-compose.yml` file is used for production. `docker-compose.override.yml` file is used for development.

I changed it.
I moved the service **mailhod** from docker-compose.yml to docker-compose.override.yml

is it ok ?

imdhemy commented:
@Moemen-Gaballah Could you please have a look on this PR #295
